postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -0,0 +1,382 @@
+        
+parser.add_argument('--maxFreq', nargs=1, default=([1.0]), metavar=('maxFreq'),
+                    help='maximum fequency of the spectrum used for the corner frequency computation' ,type=float)
+                    # maxFreq that should be sufficiently resolved by the simulation
+    
+parser.add_argument('--maxCornerFreq', nargs=1, default=([0.]), metavar=('maxCornerFreq'),
+                    help='maximum possible corner frequency (default = maxFreq)' ,type=float)
+                    # Should be specified to reduce computational costs, if the maximum expected corner frequency is
+                    # significantly less than maxFreq
+        
+parser.add_argument('--output', choices=["numpy","xdmf", "both"], default="both", 
+                    help='choose the output format')
+
+parser.add_argument('--bodywavewindow',  dest='bodywavewindow', action='store_true' , default = False, 
+                    help='cuts out a body wave window before performing the fft to mitigate the impact of surface waves')
+                    # -fault.xdmf file needed (within the same directory as -surface.xdmf or arg --faultxdmf needed)
+    
+parser.add_argument('--slipRateThreshold', nargs=1, default=([0.2]), metavar=('slipRateThreshold'),
+                    help='slip rate threshold that is used to approximate event duration and hypocenter location' ,type=float)
+                    # only needed when --bodywavewindow is active
+    
+parser.add_argument('--events', nargs=1, choices=[0,1,2], default=([0]), 
+                    help='0: xdmf file contains just one event; 1: process the first of two events; 2: process the second of two events' ,type=int) 
+                    # Needed for Ridgecrest output, where the xdmf file contains two events
+    
+parser.add_argument('--rotate',  dest='rotate', action='store_true' , default = False, 
+                    help='horizontal components are rotated w.r.t. the slip centroid') 
+                    # -fault.xdmf file needed
+    
+parser.add_argument('--outputprefix', default="data", type=str, metavar=('outputprefix'), 
+                    help='provide an outputprefix')
+
+parser.add_argument('--faultxdmf', default="input", type=str, metavar=('-fault.xdmf'), 
+                    help='provide path+filename-fault.xdmf; only needed when the path differs from the -surface.xdmf file')
+
+parser.add_argument('--serialLoading',  dest='serialLoading', action='store_true' , default = False, 
+                    help='components are loaded in serial, while calculations can use more than 1 process')
+
+parser.add_argument('--avgSWaveVelocity', nargs=1, default=([0.]), metavar=('avgSWaveVelocity'),
+                    help='provide an average S-wave velocity near the surface (otherwise it is approximated from P-arrivals)',
+                    type=float)
+args = parser.parse_args()
+
+def LoadSingleComponentParallel(component, surfaceData=True):
```

Comment:
Does it really make sense? it seems it is much faster to load on one rank the data.
If not, I would then remove this option.

Reply:
There is an optimal number of ranks (depending on the size of the loaded array), for which parallel loading is 50-100% faster. I changed the default to serial loading.